README.md
# ⚠️ Deprecated and archived. Please do not use this package.
These icons were exclusively for evernest.com and have since been moved into the evernest.com repo over this superfluous complexity.

---

# evernest-icons
- SVG icons for use in Evernest projects
- The package is [published on npm](https://www.npmjs.com/package/evernest-icons) 
- Supported sizes `16px`, `24px` and `40px`

## How to use

### Step 1 - Github
1. Create a new branch
2. Add the new icon file(s) to the `icons/in` folder with the name template `<name> <size>.svg`. 
   1. For example, if your file was a 16px SVG called `building.svg`, you would rename the file to `building 16px.svg` and place it into the `icons/in` folder. 
3. Run `yarn build`, which will automatically compress the new icons, adding them to `icons/out`, and update the types for the package.
4. Commit all the changes to your branch.

### Step 2 - Release
1. Once the PR has been approved and merged to master, `git pull` on your local master
2. Log in to NPM using `npm login`
3. Run `yarn release`. This will create the tag and version commit.
4. Run `git push --follow-tags origin master && npm publish`, which will push the new tag & commit to master and publish the new package version on NPM.


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### [0.1.18](https://github.com/everdevs/evernest-icons/compare/v0.1.17...v0.1.18) (2023-03-28)


### Features

* add 24px euro icon ([04499e8](https://github.com/everdevs/evernest-icons/commit/04499e8eb325542525800890eb3b9af5e5a4dec2))

### [0.1.17](https://github.com/everdevs/evernest-icons/compare/v0.1.16...v0.1.17) (2022-07-21)


### Features

* add filled star icon ([6f24070](https://github.com/everdevs/evernest-icons/commit/6f24070c81ed9071509e9beaab16b9470f93f1af))

### [0.1.16](https://github.com/everdevs/evernest-icons/compare/v0.1.15...v0.1.16) (2022-07-20)


### Features

* add filled star icon ([367e301](https://github.com/everdevs/evernest-icons/commit/367e30114d763a68b331e90eae820bdc9c69862d))

### [0.1.15](https://github.com/everdevs/evernest-icons/compare/v0.1.14...v0.1.15) (2022-07-20)


### Features

* add filled star icon ([4fd967f](https://github.com/everdevs/evernest-icons/commit/4fd967f8502b9e76213273182a0a3195993b0705))

### [0.1.14](https://github.com/everdevs/evernest-icons/compare/v0.1.13...v0.1.14) (2022-07-20)


### Features

* add filled star icon ([03f2caf](https://github.com/everdevs/evernest-icons/commit/03f2caf2e4d61a979310a2d43c2f6da538f946f9))

### [0.1.13](https://github.com/everdevs/evernest-icons/compare/v0.1.12...v0.1.13) (2022-07-20)


### Features

* add filled star icon ([b5f4b47](https://github.com/everdevs/evernest-icons/commit/b5f4b4713a5435c16396424d4014cd64f8a2c0af))

### [0.1.12](https://github.com/everdevs/evernest-icons/compare/v0.1.11...v0.1.12) (2022-07-19)


### Bug Fixes

* correct `buildingYear` icons ([f12c70d](https://github.com/everdevs/evernest-icons/commit/f12c70dcbb52c44154c312f71aeecada5e188230))

### [0.1.11](https://github.com/everdevs/evernest-icons/compare/v0.1.10...v0.1.11) (2022-07-19)


### Features

* add `buildingYear` icons ([2c490cd](https://github.com/everdevs/evernest-icons/commit/2c490cda65b05427329443b10832f472753505d7))

### [0.1.10](https://github.com/everdevs/evernest-icons/compare/v0.1.9...v0.1.10) (2022-07-12)


### Features

* add `tip` icon ([c6f84f9](https://github.com/everdevs/evernest-icons/commit/c6f84f9f3c4aa9e2fa0accb0f8d8d6367c531a23))

### [0.1.9](https://github.com/everdevs/evernest-icons/compare/v0.1.8...v0.1.9) (2022-03-30)


### Features

* add plus icon 24 ([c30d6fc](https://github.com/everdevs/evernest-icons/commit/c30d6fca1889f2e233987640bca975348f25433e))

### [0.1.8](https://github.com/everdevs/evernest-icons/compare/v0.1.7...v0.1.8) (2022-01-18)


### Features

* add number icons 40px ([587444b](https://github.com/everdevs/evernest-icons/commit/587444b8b85d9376c4fc5c9bf2b26fd8a62e00b6))

### [0.1.7](https://github.com/everdevs/evernest-icons/compare/v0.1.6...v0.1.7) (2022-01-18)


### Features

* add number icons ([51921d4](https://github.com/everdevs/evernest-icons/commit/51921d43f836f1113101ce72777f39bf1575d1a4))

### [0.1.6](https://github.com/everdevs/evernest-icons/compare/v0.1.5...v0.1.6) (2021-06-10)


### Features

* add twitter icon ([513895b](https://github.com/everdevs/evernest-icons/commit/513895be39b32b08b15bdd5b3911344bc5a28135))

### [0.1.5](https://github.com/everdevs/evernest-icons/compare/v0.1.4...v0.1.5) (2021-04-27)


### Bug Fixes

* fix broken icons ([055117b](https://github.com/everdevs/evernest-icons/commit/055117b42317d5f7b5d4b81f5b3570d827f8d8c1))

### [0.1.4](https://github.com/everdevs/evernest-icons/compare/v0.1.3...v0.1.4) (2021-04-27)


### Features

* add building, plot and home ([#5](https://github.com/everdevs/evernest-icons/issues/5)) ([ff3fafc](https://github.com/everdevs/evernest-icons/commit/ff3fafc8396f0f26b606744a63892ef6936441ae))

### [0.1.3](https://github.com/everdevs/evernest-icons/compare/v0.1.2...v0.1.3) (2021-03-23)


### Features

* add phone icon ([518c100](https://github.com/everdevs/evernest-icons/commit/518c100c026656f6650e1d993b323ef655b83704))

### [0.1.2](https://github.com/everdevs/evernest-icons/compare/v0.1.1...v0.1.2) (2021-02-15)


### Features

* add converted icons ([f3efe1b](https://github.com/everdevs/evernest-icons/commit/f3efe1bae4006b72488811df7d29c11fbef0cead))
* add filter and map-layers ([bc5eb9d](https://github.com/everdevs/evernest-icons/commit/bc5eb9dfb2d23dd0b10d49636b879d81ab3f01c7))
* add more icons ([43dc70c](https://github.com/everdevs/evernest-icons/commit/43dc70c764d00224a826d4dab02a0235b67054a4))
* add more icons ([d8091e9](https://github.com/everdevs/evernest-icons/commit/d8091e90243f38ae304dcc2c0b1f4f2673f4ce38))
* add more icons as source ([4d79eba](https://github.com/everdevs/evernest-icons/commit/4d79eba5119f68c77c4e401624df5ae987b05753))
* optimize icons ([32f5058](https://github.com/everdevs/evernest-icons/commit/32f5058582b5f29ac8dddec2a015b8682aa768bb))

### [0.1.1](https://github.com/everdevs/evernest-icons/compare/v0.1.0...v0.1.1) (2021-02-08)


### Features

* add icons ([c286aa2](https://github.com/everdevs/evernest-icons/commit/c286aa2695430afab91f0512fd6506396e8751db))

## 0.1.0 (2021-02-04)


### Features

* add close icon ([f313a9d](https://github.com/pixelass/evernest-icons/commit/f313a9dc0ac05a592989a8acab76b415405f3491))
* add close icons ([91172a8](https://github.com/pixelass/evernest-icons/commit/91172a8c347c11a8807d756ef8474f221d93e385))
* add icons and converter ([2fc0498](https://github.com/pixelass/evernest-icons/commit/2fc0498c3fe84756ffca4f059f10c1ba86ed4371))
* add javascript file ([31011dc](https://github.com/pixelass/evernest-icons/commit/31011dcdefa23bfa6f9c735652690c41399ed64a))
* don't remove xmlns ([a45966e](https://github.com/pixelass/evernest-icons/commit/a45966e9eb4f5e46cfa060651302b8eb85d7bb88))
* export d attributes ([ba4b30d](https://github.com/pixelass/evernest-icons/commit/ba4b30dc1bd839af7a5206795e012ea7c90fb7f4))

rename.js
const path = require("path");
const pify = require("pify");
const globby = require("globby");
const { paramCase } = require("param-case");
const { camelCase } = require("camel-case");
const transliterate = require("@sindresorhus/transliterate");
const mkdirp = require("mkdirp");
const { readFile, writeFile } = pify(require("fs"));

(async () => {
	const paths = await globby(["./icons/optimized"]);
	const types = { 16: {}, 24: {}, 40: {} };
	const pathD = { 16: {}, 24: {}, 40: {} };
	const files = { 16: {}, 24: {}, 40: {} };
	await Promise.all(
		paths.map(async file => {
			const { name } = path.parse(file);
			const [, iconName, size] = /(.*) (\d+)px/.exec(name);
			const camelCased = camelCase(transliterate(iconName));
			const content = await readFile(file, "utf-8");
			const [, d] = /d="([\w\d\-\s\.]+)"/.exec(content);
			types[size][camelCased] = "string";
			pathD[size][camelCased] = d;
			files[size][camelCased] = content;
			// console.log(`Writing: ./icons/out/${paramCased}.svg`);
			// return writeFile(`./icons/out/${paramCased}.svg`, content);
		})
	);
	Object.entries(files).forEach(entry => {
		const [size, icons] = entry;
		Object.entries(icons).forEach(async icon => {
			const [name, content] = icon;
			await mkdirp(`./icons/out/${size}`).catch(err => {
				console.error("mkdirp", err);
			});
			writeFile(`./icons/out/${size}/${name}.svg`, content).catch(err => {
				console.error(err);
			});
		});
	});
	console.log(`Writing: ./icons/path-d.json`);
	writeFile(`./icons/path-d.json`, JSON.stringify(pathD, null, 4));
	console.log(`Writing: ./icons/path-d.js`);
	writeFile(`./icons/path-d.js`, `module.exports = ${JSON.stringify(pathD, null, 4)}`);
	console.log(`Writing: ./icons/path-d.d.ts`);
	writeFile(
		`./icons/path-d.d.ts`,
		`interface Icons ${JSON.stringify(types, null, 4)
			.replace(/"string"(,)?/g, "string;")
			.replace(/}(,)?/g, "};")
			.replace(/^};$/gm, "}")}\nexport default Icons;`
	);
})();


convert.js
const path = require("path");
const pify = require("pify");
const globby = require("globby");
const { camelCase } = require("camel-case");
const transliterate = require("@sindresorhus/transliterate");
const mkdirp = require("mkdirp");
const SVGO = require("svgo");
const deindent = require("de-indent");
const svgoConfig = require("./svgo.json");

const svgo = new SVGO(svgoConfig);

const { readFile, writeFile } = pify(require("fs"));
const rimraf = pify(require("rimraf"));

// These files and directories are written and should be deleted
const writePaths = [
	"./icons/out",
	"./icons/path-d.json",
	"./icons/path-d.js",
	"./icons/path-d.d.ts",
];

const createTypeDeclarations = names => {
	const namesUnion = names.map(name => `"${name}"`).join(" | ");
	const typeDeclarations = deindent(`
		export enum IconSize {
			s = "16",
			m = "24",
			l = "40",
		}
		export type IconName = ${namesUnion}
		export type IconCollection = {
			[key in IconName]: string;
		}
		export type Icons = {
			[key in IconSize]: IconCollection;
		}
		export declare const icons: Icons
		export default icons
	`);
	return typeDeclarations;
};

(async () => {
	// Delete all writePaths
	await Promise.all(writePaths.map(writePath => rimraf(writePath)));
	const paths = await globby(["./icons/in/*.svg"]);
	const sortedPaths = paths.sort((a, b) => {
		if (a > b) {
			return 1;
		}
		if (a < b) {
			return -1;
		}
		return 0;
	});
	// Provide objects that will be filled with data
	const names = new Set([]);
	const pathD = { 16: {}, 24: {}, 40: {} };
	const files = { 16: {}, 24: {}, 40: {} };
	// Fill the objects with the correct data
	await Promise.all(
		sortedPaths.map(async file => {
			const { name } = path.parse(file);
			const [, iconName, size] = /(.*) (\d+)px/.exec(name);
			// Names should be camelcased
			const camelCased = camelCase(transliterate(iconName));
			const rawSVG = await readFile(file, "utf-8");
			// SVGs should be optimized
			const { data: optimizedSVG } = await svgo
				.optimize(rawSVG, { path: file })
				.catch(err => {
					console.error(err);
				});
			try {
				// Extract the d attribute from the path
				const [, d] = / d="([\w\d\-\s.,]+)"/.exec(optimizedSVG);
				// Add data to objects
				names.add(camelCased);
				pathD[size][camelCased] = d;
				files[size][camelCased] = optimizedSVG;
			} catch (err) {
				console.error(err);
			}
		})
	);
	// Write all SVGs as optimized files
	// Files are sorted by their
	Object.entries(files).forEach(entry => {
		const [size, icons] = entry;
		Object.entries(icons).forEach(async icon => {
			const [name, svg] = icon;
			// Create the directory before we attempt to write the file
			await mkdirp(`./icons/out/${size}`).catch(err => {
				console.error("mkdirp", err);
			});
			writeFile(`./icons/out/${size}/${name}.svg`, svg).catch(err => {
				console.error(err);
			});
		});
	});

	// Find missing icons
	const iconNames = Object.entries(pathD).reduce(
		(previousValue, [key, value]) => ({ ...previousValue, [key]: Object.keys(value) }),
		{}
	);
	const { 16: s, 24: m, 40: l } = iconNames;
	const sMisses = m.filter(name => !s.includes(name));
	const lMisses = m.filter(name => !l.includes(name));
	const missingIcons = `# Missing icons\n\n\n## 16px\n\n${sMisses.map(miss => `* ${miss}`).join("\n")}\n\n\n## 40px\n\n${lMisses.map(miss => `* ${miss}`).join("\n")}\n`;
	// Create JSON
	writeFile(`./icons/missing.md`, missingIcons).catch(err => {
		console.error(err);
	});
	// Create JSON
	writeFile(`./icons/path-d.json`, JSON.stringify(pathD, null, 4)).catch(err => {
		console.error(err);
	});

	// Create JavaScript
	writeFile(
		`./icons/path-d.js`,
		deindent(`
		'use strict';
		Object.defineProperty(exports, '__esModule', { value: true });
		const IconSize = {
			s: "16",
			m: "24",
			l: "40",
		}
		const icons = ${JSON.stringify(pathD, null, 4)};
		exports.icons = icons
		exports.IconSize = IconSize
		`)
	).catch(err => {
		console.error(err);
	});

	// Create Type definitions
	// replace quotes and commas for validity
	const typeDeclarations = createTypeDeclarations(Array.from(names));
	writeFile(`./icons/path-d.d.ts`, typeDeclarations).catch(err => {
		console.error(err);
	});
})();


icons/path-d.d.ts

export enum IconSize {
	s = "16",
	m = "24",
	l = "40",
}
export type IconName = "2" | "3" | "360" | "4" | "1" | "5" | "6" | "7" | "8" | "9" | "academy" | "arrowRight" | "bell" | "building" | "buildingYear" | "chevronDown" | "chevronLeft" | "chevronRight" | "chevronUp" | "click" | "delete" | "filter" | "handshake" | "euro" | "home" | "facebook" | "info" | "instagram" | "landArea" | "list" | "linkedIn" | "livingSpace" | "location" | "logout" | "mail" | "map" | "mapLayers" | "mobileDevice" | "money" | "onboarding" | "person" | "phone" | "phonebook" | "plot" | "plusBox" | "plus" | "star" | "rooms" | "starFilled" | "tip" | "twitter" | "video" | "world" | "xing"
export type IconCollection = {
	[key in IconName]: string;
}
export type Icons = {
	[key in IconSize]: IconCollection;
}
export declare const icons: Icons
export default icons
